xapi-model/src/main/java/dev/learning/xapi/model/Statement.java

```diff
@@ -127,6 +134,45 @@ public static class Builder {
 
     // This static class extends the lombok builder.
 
+    /**
+     * Special build method for signing and building a {@link Statement}.
+     * <p>
+     * An signature attachment is automatically added to the Statement's attachments.
+     * </p>
+     *
+     * @param privateKey a {@link PrivateKey} for signing the {@link Statement}.
+     * @return an immutable, signed {@link Statement} object.
+     * @see <a href=
+     *      "https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#26-signed-statements">
+     *      Signed Statements</a>
+     */
+    public Statement sign(PrivateKey privateKey) {
+      Map<String, Object> claims = new HashMap<>();
+
+      // Put only the significant properties into the signature payload
+      // https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#statement-comparision-requirements
+      claims.put("actor", this.actor);
+      claims.put("verb", this.verb);
+      claims.put("object", this.object);
+      claims.put("result", this.result);
+      claims.put("context", this.context);
+
+      String token = Jwts.builder().setClaims(claims).signWith(privateKey, SignatureAlgorithm.RS512)
```

Comment:
catch some sort of class not found exception and give a sensible error message. 

Please test.